openwisp_radius/management/commands/base/convert_called_station_id.py

```diff
@@ -0,0 +1,111 @@
+        return raw_management_info
+
+    def _get_openvpn_routing_info(self, host, port=7505, password=None):
+        raw_info = self._get_raw_management_info(host, port, password)
+        parsed_info = openvpn_status.parse_status(raw_info)
+        return parsed_info.routing_table
+
+    def handle(self, *args, **options):
+        for org_slug, config in CALLED_STATION_IDS.items():
+            routing_dict = {}
+            for openvpn_config in config['openvpn_config']:
+                try:
+                    routing_dict.update(
+                        self._get_openvpn_routing_info(
+                            openvpn_config['host'],
+                            openvpn_config.get('port', 7505),
+                            openvpn_config.get('password', None),
+                        )
+                    )
+                except ConnectionRefusedError:
+                    logger.error(
+                        'Unable to establish telnet connection to '
+                        f'{openvpn_config["host"]} on {openvpn_config["port"]}. '
+                        'Skipping!'
+                    )
+                except openvpn_status.ParsingError as error:
+                    logger.error(
+                        'Unable to parse information received from '
+                        f'{openvpn_config["host"]}. ParsingError: {error}. Skipping!',
+                    )
+            if not routing_dict:
+                logger.info(
+                    'Could not fetch any routing information for '
+                    f'organization with "{org_slug}" slug'
+                )
+                continue
+
+            qs = RadiusAccounting.objects.filter(
+                organization__slug=org_slug,
+                called_station_id__in=config['unconverted_macs'],
+            )
+            for radius_session in qs.iterator():
+                try:
+                    common_name = routing_dict[
+                        radius_session.calling_station_id
+                    ].common_name
+                    mac_address = RE_VIRTUAL_ADDR_MAC.search(common_name)[0]
+                    radius_session.called_station_id = mac_address.replace(':', '-')
+                except KeyError:
+                    logger.warn(
+                        'Failed to find routing information for '
+                        f'{radius_session.session_id}. Skipping!'
+                    )
+                except (TypeError, IndexError):
+                    logger.warn(
+                        f'Failed to find a MAC address in "{common_name}". '
+                        f'Skipping {radius_session.session_id}!'
+                    )
+                else:
+                    radius_session.save()
+
+
+# monkey patching for openvpn_status begins
+def parse_virtual_address(virtual_address):
+    return openvpn_status.utils.parse_vaddr(virtual_address.split('@')[0])
+
+
+openvpn_status.utils.DATETIME_FORMAT_OPENVPN = u'%Y-%m-%d %H:%M:%S'
```

Comment:
this is quite brutal, because it would mean this command would only work on pfsense. I think we should add a way to specify this via configuration so that if a specific flag is enabled, this is done.